workspaces/optic-engine/tests/shape_diff.rs

```diff
@@ -436,8 +436,9 @@ fn can_diff_optional() {
     &shape_id,
   );
 
+  dbg!(&results);
   assert_debug_snapshot!("can_diff_optional__results", results);
-  assert_eq!(results.len(), 2); //@BUG: this should be 1, right? and the trail looks wrong
+  assert_eq!(results.len(), 1);
```

Comment:
Turns out this was already a bug we knew about, with this test isolating the same case as the new test.

Reply:
The diffing should not have emitted a diff result for the OptionalTrail in the 3rd array item (empty object), so to me this indicated diff logic as opposed to list_trail_choices

Reply:
It doesn't do that? Both diffs described a diff with the second array item being a number instead of a string, just with two different shape trails.